geode-core/src/main/java/org/apache/geode/security/StreamAuthenticator.java

```diff
@@ -46,6 +44,14 @@ void receiveMessage(InputStream inputStream, OutputStream outputStream,
   boolean isAuthenticated();
 
   /**
+   * Return an authorization object which can be used to determine which permissions this stream has
+   * according to the provided securityManager.
+   *
+   * Calling this before authentication has succeeded may result in a null return object.
```

Comment:
Could we indicate this by using a Optional? That way we also force the consumer to deal with that case.

Reply:
I kind of prefer a comment or even annotation saying nullable -- optionals themselves can be null, and have some cost.

Reply:
How am I gonna see the comment as a user of this? I think it's unreasonable to assume that every consumer of the class is gonna look at its code.

Reply:
`Optional` doesn't really fix the problem, just introduces more GC pressure :-)

Wouldn't it be better to throw `IllegalStateException` when the precondition is not met?

Reply:
We'll change this to throw AuthenticationRequiredException if authentication has not yet been performed.